### PR TITLE
Added Rubocop

### PR DESCRIPTION
**Pull request by nikhilgupta1211**

### CODE REVIEW

**Comment by bgeuken** [comment]
@nikhilgupta1211 JFYI, I just merged #54. You'll need to rebase your branch.

File: Gemfile [hunk 1]
```diff
@@ -26,56 +28,58 @@ group :doc do
...
 group :development do
-  gem "yard-activerecord"
-  gem 'state_machines-yard'
+  gem 'redcarpet', require: false
   gem 'ruby-graphviz'
-  gem "redcarpet", :require => false
+  gem 'state_machines-yard'
+  gem 'yard-activerecord'
+  # for static code analisys
+  gem 'rubocop', '~> 0.49.1', require: false
```
Comment (bgeuken) [inline Gemfile]: Why did you add the 'require: false' here?

File: Gemfile [hunk 2]
```diff
@@ -26,56 +28,58 @@ group :doc do
...
+gem 'inherited_resources'
+gem 'kaminari'
+gem 'localized_country_select'
 gem 'prawn', '~> 0.13.0'
 gem 'prawn_rails'
-gem 'axlsx_rails'
-gem 'clockwork'
+gem 'ransack'
+gem 'show_for'
+gem 'simple_form'
```
Comment (bgeuken) [inline Gemfile]: Did you reorder the gems or was that rubocop? And why did you change the order, if you did?

File: Gemfile (hunk 1 above)
Comment (bgeuken) [inline Gemfile]: Ah, nevermind

Comment (nikhilgupta1211) [inline Gemfile]: Because we don't need it to be required in the code but to only run it from command line

File: Gemfile (hunk 2 above)
Comment (nikhilgupta1211) [inline Gemfile]: rubocop autocorrect reordered the gems

**Comment by bgeuken** [comment]
@nikhilgupta1211 The tsp page states that we support Ruby >= 1.9. I am currently try to find out if this is still up to date. But I assume that we have to support at least v2.0 or v2.1. So we can not use features like [&.](https://github.com/openSUSE/travel-support-program/pull/55/files#diff-88abeec4ba9bdccfec0d0ef524bcd854R10).

On my development machine I am using ruby 2.2.6 and currently I get 19 test failures with this branch.

**Comment by nikhilgupta1211** [comment]
Actually rubocop need ruby version >= 2.0.0, so which version should we use ?

**Comment by bgeuken** [comment]
> nikhilgupta1211 commented an hour ago
>
> Actually rubocop need ruby version >= 2.0.0, so which version should we use ?

I any case something below 2.3. There might be a way to configure this in rubocop... at least for the autogenerated code. I would check that first.

File: .rubocop.yml [hunk 3]
```diff
@@ -12,3 +12,207 @@ AllCops:
     - 'bundle/**/*'
     - 'config/**/*'
     - 'bin/*'
+
```
Comment (bgeuken) [inline .rubocop.yml]: Nice that you split the auto-correct changes in a separate commit :+1:

**Comment by nikhilgupta1211** [comment]
we can set the ruby version like this
```
AllCops:
  TargetRubyVersion: 2.0
```

**Comment by bgeuken** [comment]
Yep. Please do that. After that you might have to run the auto-correct again.

**Comment by nikhilgupta1211** [comment]
ok :)

**Comment by nikhilgupta1211** [comment]
I have set the ruby version to 2.1.0

**Comment by bgeuken** [comment]
@nikhilgupta1211 Just spoke with Ancor. For now we have to stick with ruby 1.9 syntax. I would advise to start this from scratch, remove the auto corrects, add target version to 1.9 and then redo what you did before.

**Comment by bgeuken** [comment]
Nice work :+1:
